libs/hdf-converters/src/neuvector-mapper.ts

```diff
@@ -0,0 +1,311 @@
+  package_name: string;
+  package_version: string;
+  fixed_version: string;
+  link: string;
+  score_v3: number /* float32 */;
+  vectors_v3: string;
+  published_timestamp: number /* int64 */;
+  last_modified_timestamp: number /* int64 */;
+  cpes?: string[];
+  cves?: string[];
+  feed_rating: string;
+  in_base_image?: boolean;
+  tags?: string[];
+};
+
+type RESTScanModule = {
+  name: string;
+  file: string;
+  version: string;
+  source: string;
+  cves?: RESTModuleCve[];
+  cpes?: string[];
+};
+
+type RESTModuleCve = {
+  name: string;
+  status: string;
+};
+
+type RESTBenchItem = {
+  level: string;
+  evidence?: string;
+  location?: string;
+  message: string[];
+  group?: string;
+} & RESTBenchCheck;
+
+type RESTBenchCheck = {
+  test_number: string;
+  category: string;
+  type: string;
+  profile: string;
+  scored: boolean;
+  automated: boolean;
+  description: string;
+  remediation: string;
+  tags?: string[]; // Tags provide list of compliance that related to the cis test item.
+  tags_v2?: Record<string, any /* share.TagDetails */>; // TagsV2 provide compliance details for each compliance tag
+};
+
+type RESTScanSecret = {
+  type: string; // the secret description
+  evidence: string; // found in a cloaked string
+  path: string; // file path
+  suggestion: string;
+};
+
+type RESTScanSetIdPerm = {
+  type: string; // the set id descriptions
+  evidence: string; // file attributes
+  path: string; // file path
+};
+
+type RESTScanSignatureInfo = {
+  verifiers?: string[];
+  verification_timestamp: string;
+};
+
+type RESTScanReport = {
+  vulnerabilities: RESTVulnerability[];
+  modules?: RESTScanModule[];
+  checks?: RESTBenchItem[];
+  secrets?: RESTScanSecret[];
+  setid_perms?: RESTScanSetIdPerm[];
+  envs?: string[];
+  labels?: Record<string, string>;
+  cmds?: string[];
+  signature_data?: RESTScanSignatureInfo;
+};
+
+type RESTScanRepoReport = {
+  verdict?: string;
+  image_id: string;
+  registry: string;
+  repository: string;
+  tag: string;
+  digest: string;
+  size: number /* int64 */;
+  author: string;
+  base_os: string;
+  created_at: string;
+  cvedb_version: string;
+  cvedb_create_time: string;
+  layers: RESTScanLayer[];
+} & RESTScanReport;
+
+type RESTScanLayer = {
+  digest: string;
+  cmds: string;
+  vulnerabilities: RESTVulnerability[];
+  size: number /* int64 */;
+};
+
+export type NeuvectorScanJson = {
+  report: RESTScanRepoReport;
+  error_message: string;
+};
+
+const CWE_NIST_MAPPING = new CweNistMapping();
+
+function cweTags(description: string): string[] | undefined {
+  const regex = /CWE-\d{3}/g;
+  return description?.match(regex)?.flat();
+}
+
+function nistTags(cweTags: string[] | undefined): string[] {
+  const identifiers =
+    cweTags?.map((tag: string) => tag.match(/\d{3}/g)?.[0] ?? [])?.flat() ?? [];
+  return CWE_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_UPDATE_REMEDIATION_NIST_TAGS
+  );
+}
+
+function ghsaTag(name: string): string | undefined {
```

Comment:
I can remove them, but just wondering, are GHSAs and RHSAs unnecessary to list in the tags because CVEs/CWEs are more precise in what information they display or that they're primarily mapped in heimdall2?

Reply:
usually cves or other unique identifiers end up being control.ids.  cwes and nist tags are a) more general purpose information and b) used for doing mappings.